pkg/state/appender.go

```diff
@@ -384,6 +384,82 @@ type appendTxParams struct {
 	initialisation   bool
 }
 
+func (a *txAppender) guessEthereumTransactionKind(ethTx *proto.EthereumTransaction, params *appendTxParams) (proto.EthereumTransactionKind, error) {
+	if ethTx.Data() == nil {
+		return proto.NewEthereumTransferWavesTx(), nil
+	}
+
+	selectorBytes := ethTx.Data()
+	selector, err := ethabi.NewSelectorFromBytes(selectorBytes[:4])
```

Comment:
You should check that `len(ethTx.Data()) >= 4` or `len(ethTx.Data()) >= ethabi.SelectorSize`.